angular/playground/projects/ng-entity-service/src/lib/ng-entity.service.ts

```diff
@@ -72,123 +84,121 @@ export class NgEntityService<S extends EntityState = any> extends EntityService<
+      entityId,
       storeName: this.store.storeName
     });
 
-    return this.http[method.toLowerCase()](url, _config).pipe(
-      mapResponse(_config),
+    return this.http.request(method, url, conf).pipe(
+      mapResponse(conf),
       tap((data: any) => {
-        if (isSingle) {
-          this.store.upsert(idOrConfig as getIDType<S>, data);
-        } else {
-          if (_config.append) {
-            this.store.add(data);
-          } else if (_config.upsert) {
-            this.store.upsertMany(data);
+        if (!conf.skipWrite) {
+          if (isSingle) {
+            this.store.upsert(entityId, data);
           } else {
-            this.store.set(data);
+            if (conf.append) {
+              this.store.add(data);
+            } else if (conf.upsert) {
+              this.store.upsertMany(data);
+            } else {
+              this.store.set(data);
+            }
           }
         }
 
         this.dispatchSuccess({
           method,
           payload: data,
-          successMsg: _config.successMsg
+          successMsg: conf.successMsg
         });
       }),
-      catchError(error => this.handleError(method, error, _config.errorMsg)),
+      catchError(error => this.handleError(method, error, conf.errorMsg || '')),
       finalize(() => {
         this.loader.dispatch({
           method,
           loading: false,
           storeName: this.store.storeName
         });
       })
-    ) as Observable<T>;
+    );
   }
 
   /**
-   *
    * Add a new entity - Creates a POST request
    *
-   * service.add(entity)
-   * service.add(entity, config)
-   *
+   * @example
+   * service.add(entity).subscribe()
+   * service.add(entity, config).subscribe()
    */
-  add<T>(entity: getEntityType<S>, config?: HttpConfig & Pick<AddEntitiesOptions, 'prepend'> & Msg): Observable<T> {
+  add<T>(entity: getEntityType<S>, config?: HttpAddConfig<T>): Observable<T> {
+    const url = this.resolveUrl(config);
     const method = this.getHttpMethod(HttpMethod.POST);
 
     this.loader.dispatch({
       method,
       loading: true,
       storeName: this.store.storeName
     });
-    return this.http[method.toLowerCase()](this.resolveUrl(config), entity, config).pipe(
+
+    const configWithBody = { ...config, ...{ body: entity } };
+
+    return this.http.request(method, url, configWithBody).pipe(
       mapResponse(config),
-      tap((entity: any) => {
-        this.store.add(entity, config);
+      tap((responseEntity: any) => {
+        if (!config || (config && !config.skipWrite)) {
+          this.store.add(responseEntity, config);
+        }
         this.dispatchSuccess({
           method,
-          payload: entity,
+          payload: responseEntity,
           successMsg: config && config.successMsg
         });
       }),
-      catchError(error => this.handleError(method, error, config && config.errorMsg)),
+      catchError(error => this.handleError(method, error, (config && config.errorMsg) || '')),
```

Comment:
There is no need to make the ` || ''` check in each method. Do it once in `handleError`